quic/s2n-quic-core/src/recovery/bandwidth/estimator.rs

```diff
@@ -82,6 +82,20 @@ impl core::ops::Mul<Duration> for Bandwidth {
     }
 }
 
+/// Divides a count of bytes represented as a u64 by the given `Bandwidth`
+///
+/// Since `Bandwidth` is a rate of bytes over a time period, this division
+/// results in a `Duration` being returned, representing how long a path
+/// with the given `Bandwidth` would take to transmit the given number of
+/// bytes.
+impl core::ops::Div<Bandwidth> for u64 {
+    type Output = Duration;
+
+    fn div(self, rhs: Bandwidth) -> Self::Output {
+        Duration::from_micros(self * MICRO_BITS_PER_BYTE / rhs.bits_per_second)
```

Comment:
Actually, it might just be better to leave it as is and allow for the Panic that would occur if one attempted to divide by zero, as that is an invalid request anyway. I'll be adding additional testing/fuzzing to ensure we don't encounter these scenarios.

Reply:
In Java, I would have thrown an `BadArgumentException` or something similar during initialisation. Does it make sense to do the same in rust? I guess custom errors would be the closest

Reply:
I'm wondering if it would be more efficient to store `nanos_per_byte` or `nanos_per_bit` rather than `bits_per_second`. This would allow you to rewrite a lot of the divisions into multiplications, remove the majority of the panics, and preserve accuracy:

[nanos-per-byte](https://godbolt.org/z/j8K89vvx3)
[nanos-per-bit](https://godbolt.org/z/Kf4o8Eo98)

Reply:
I'll try that in a followup PR

Reply:
https://github.com/aws/s2n-quic/pull/1439